src/main/java/crawlercommons/filters/basic/BasicURLNormalizer.java

```diff
@@ -277,13 +296,13 @@ public String filter(String urlString) {
     private static URL parseStringToURL(String urlString) {
         URL url = null;
         try {
-            url = new URL(urlString);
-        } catch (MalformedURLException e) {
+            url = new URI(urlString).toURL();
+        } catch (MalformedURLException | URISyntaxException | IllegalArgumentException e) {
             if (!hasSchemePattern.matcher(urlString).find()) {
                 // no protocol/scheme : try to prefix http://
                 try {
-                    url = new URL("http://" + urlString);
```

Comment:
Ok, here might be the origin for the failing unit test: "http://" is added in front of "http:".